api/wasm/cpp/proxy_wasm_api.h

```diff
@@ -427,6 +412,21 @@ struct RegisterContextFactory {
                                   RootFactory root_factory = nullptr, StringView root_id = "");
 };
 
+inline WasmDataPtr getConfiguration() {
+  const char* value_ptr = nullptr;
+  size_t value_size = 0;
+  CHECK_RESULT(proxy_getConfiguration(&value_ptr, &value_size));
+  return std::make_unique<WasmData>(value_ptr, value_size);
+}
+
+inline std::pair<uint32_t, WasmDataPtr> getStatus() {
```

Comment:
And for generic WasmResult != Ok.  It would be nice to have an addition details message available for debugging.

I see this more googl3 Status which is completely generic. It has error namespaces, perhaps you would be interested in adding that?  e.g. GrpcErrors, HttpErrors, WasmErrors ?